### PR TITLE
10639 geometry changed it s shape after moving around the map

### PR DESCRIPTION
**Pull request by IlyaIzr**
Nebula GL used Translate Mode to move features on map
This mode used turfTransformTranslate(), giving it any geometry at all
And that caused serious deformations when moving polygons to poles

So I implemented a solution that moves polygons center first and then builds polygon around it. That way geometry distortion is significantly smaller and not really visible

### /CustomTransformMode.ts is a copy of nebula module with 1 import change
(to use local translation mode)

### CODE REVIEW

File: src/core/draw_tools/map-daw-tools/customDrawModes/translateFunction.test.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  expect(result.geometry.coordinates[0][1]).toBeTypeOf('object');
+  expect(result.geometry.coordinates[0][1].length).toBeDefined();
+  expect(result.geometry.coordinates[0][2]).toBeTypeOf('object');
+  expect(result.geometry.coordinates[0][2].length).toBeDefined();
+
+  expect(result.geometry.coordinates[0][3]).toBeUndefined();
+  expect(result.geometry.coordinates[1]).toBeUndefined();
+});
+
+test('Polygon coordinates in right format', () => {
```
Comment (Akiyamka) [inline src/core/draw_tools/map-daw-tools/customDrawModes/translateFunction.test.ts:114]: Tests should have different names, How you understand from log what test is fail? Be more specific

File: src/core/draw_tools/map-daw-tools/customDrawModes/CustomModifyMode.ts [hunk 2]
```diff
@@ -77,7 +78,7 @@ export class CustomModifyMode extends GeoJsonEditMode {
           this._submodesCache[submodeName] = new ModifyMode();
           break;
         case 'Transform':
-          this._submodesCache[submodeName] = new TransformMode();
+          this._submodesCache[submodeName] = new CustomTransformMode();
```
Comment (Akiyamka) [inline src/core/draw_tools/map-daw-tools/customDrawModes/CustomModifyMode.ts:81]: Currently the only  way to understand difference between TransformMode and CustomTransformMode in line by line soruce code comparing. Need more details

**Comment by Komzpa** [comment]
Can we fix this in Nebula upstream instead?

**Comment by Akiyamka** [comment]
I think instead of making a custom mod we should make a fix in the nebula. After all, we want to get fixes from them too

**Comment by IlyaIzr** [comment]
After preparing upstream MR to Nebula i found out there was possibly a way already to do the lossless translation. I'm trying to configure it now, as it's almost not documented and then compare to my solution

**Comment by IlyaIzr** [comment]
Checked Nebula implemented ' without distortion ' translation. It just moves shape without any transormation according to projection
image

**Comment by Komzpa** [comment]
This is not desired behavior as the shape is not becoming larger when going towards the pole.

**Comment by IlyaIzr** [comment]
Pull request to Nebula: https://github.com/uber/nebula.gl/pull/772

**Comment by Komzpa** [comment]
@IlyaIzr is there a screenshot/gif on what happens when you move geometry to far north? (it has to grow larger)

**Pull request closed without merging**